enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,303 @@
+
+```go
+
+type FeatureSet string
+
+var (
+  ...
+  // LegacyJenkinsPipeline enables the legacy JenkinsPipeline build strategy
+  LegacyJenkinsPipeline = "LegacyJenkinsPipeline"
+)
+
+...
+
+var FeatureSets = map[FeatureSet]*FeatureGateEnabledDisabled{
+  ...
+  LegacyJenkinsPipeline: newDefaultFeatures().
+    with("BuildJenkinsPipelineStrategy").
+    toFeatures(),
+}
+
+```
+
+The operator for openshift-controller-manager will be enhanced to watch the cluster feature gates.
+If the cluster has the `BuildJenkinsPipelineStrategy` feature gate enabled, this gate will be passed to openshift-controller-manager's configuration.
+Openshift-controller-manager will then enable the strategy and allow JenkinsPipline builds to proceed as normal.
+
+#### Bootstrap Behavior
+
+When the `BuildJenkinsPipelineStrategy` feature gate is introduced, the openshift-controller-manager-operator will query all `BuildConfig` objects to see if any utilize the JenkinsPipeline strategy.
+If any BuildConfig uses the JenkinsPipline strategy, the LegacyJenkinsPipeline feature set will be turned on.
+
+The bootstrap check will only run if the default feature set is configured for the cluster.
+A status condition will indicate if this bootstrap check has completed.
+Once the bootstrap check is complete, it should not be run again.
+This protections will ensure that we do not overwrite feature sets that are enabled as a day 2 action, or overwrite pre-existing feature sets that allow updgrades.
+
+#### oc <new|start>-build
+
+`oc new-build` should print a warning message if the generated/referenced BuildConfig uses the Jenkins pipeline strategy.
+This message should indicate that the cluster must have the LegacyJenkinsPipeline feature set enabled.
+Similarly, `oc start-build` should print a warning message if a build was started with the Jenkins pipeline strategy indicating that the cluster must have the LegacyJenkinsPipeline feature set enabled.
+
+#### Web Console
+
+User will continue to be able to deploy Jenkins on OpenShift by instantiating our provided Templates.
+As long as the Jenkins Pipeline strategy can be run on a fully supported cluster (i.e upgrades are allowed), existing console integrations with Jenkins should remain in place.
+
+#### Telemetry
+
+The feature set will take advantage of existing telemetry tooling.
+Clusters that enable the LegacyJenkinsPipeline feature set will be automatically reported to Telemeter.
+
+#### Documentation
+
+After Jenkins is removed from the payload, documentation related to Jenkins images will need to add mirroring instructions for disconnected clusters.
+Release notes will also need to inform admins that Jenkins images will need to be mirrored separately from the OpenShift payload.
+
+Documentation related to the Jenkins Pipeline build strategy will need to inform admins that the `LegacyJenkinsPipeline` feature set needs to be enabled for builds to succeed.
+
+### Risks and Mitigations
```

Comment:
i'd like to see more detail around the imagestream transition from "imagestream that refs payload content" to "imagestream that refs external content"

For a connected cluster, this isn't a big deal and should be basically transparent when the cluster upgrades.

For a disconnected cluster it means:
1) on install, the imagestream won't import (if the image wasn't mirrored, which it won't be if they just mirrored the payload).  That's ok, we can document/talk through that
2) On upgrade what happens?  the imagestream gets updated and now points to some external registry and the imagestream can't be imported....  does it keep limping along on the last imported SHA (the one from the previous OCP release payload)?  And how bad a problem is that if that's what happens?  This is the scenario i'd be most worried about.

Reply:
Split this out into the new proposal - #751.

On upgrade the status will reflect the last imported SHA until the new image is mirrored. Eventually the Samples operator will fire an alert indicating that the Jenkins imagestream(s) have failed to import.